keras_nlp/src/models/stable_diffusion_3/mmdit_block.py

```diff
@@ -240,7 +241,9 @@ def _compute_attention(self, query, key, value):
             query, ops.cast(self._inverse_sqrt_key_dim, query.dtype)
         )
         attention_scores = ops.einsum(self._dot_product_equation, key, query)
-        attention_scores = ops.nn.softmax(attention_scores, axis=-1)
+        original_dtype = attention_scores.dtype
+        attention_scores = self.softmax(attention_scores)
+        attention_scores = ops.cast(attention_scores, original_dtype)
```

Comment:
can we just do `ops.cast(attention_scores, self.compute_dtype)` here? slightly clearer